src/pykeen/training/training_loop.py

```diff
@@ -107,12 +107,13 @@ def __init__(
             training and batch size during evaluation with regards to the hardware at hand.
         """
         self.model = model
-        self.triples_factory = triples_factory
         self.optimizer = optimizer
         self.training_instances = None
         self.losses_per_epochs = []
         self.automatic_memory_optimization = automatic_memory_optimization
 
+        logger.debug("we don't really need the triples factory: %s", triples_factory)
```

Comment:
we need to keep this in the interface because the sLCWA loop uses it to construct the negative sampler

Reply:
Could we defer the instantiation of the negative sampler to the actual `train` call?

Reply:
yes but that's the job for next PR in the series ;)

Reply:
Ultimately we won't be able to get around some of this stuff - the negative sampler holds a filter which has to be able to hold the entire set of training triples in some cases

Reply:
> the negative sampler holds a filter which has to be able to hold the entire set of training triples in some cases

This only happens in the (rare) case of enabling filtering for negative sampler.